daemon/snap.go

```diff
@@ -172,6 +173,9 @@ func mapSnap(localSnap *snap.Info, active bool, remoteSnap *snap.Info) map[strin
 		if summary == "" {
 			summary = remoteSnap.Summary()
 		}
+		if snapID == "" {
+			snapID = remoteSnap.SnapID
```

Comment:
I'm really glad this coalescing logic is going away with the /v2/find vs. /v2/snaps split. This is one case that looks awkward and potentially dangerous. We're guessing the primary key which we use to make critical decisions surrounding the identity of the item.